### PR TITLE
[REVIEW] ignore list dtypes for select_dtype, similar to pandas

### LINKED ISSUE #6919: [BUG]Select dtypes fails if list dtypes present in dataframe
**Describe the bug**
Select dtypes fails if list dtypes present in dataframe 

**Steps/Code to reproduce bug**
```python
import cudf
df = cudf.DataFrame({'int_col':[0,1,2],'list_col':[[1,2],[3,4],[5,6]]})
df.select_dtypes('int64').columns
```
```python
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-3-1632dc06c94a> in <module>
      1 import cudf
      2 df = cudf.DataFrame({'int_col':[0,1,2],'list_col':[[1,2],[3,4],[5,6]]})
----> 3 df.select_dtypes('int64').columns

/nvme/0/vjawa/vjawa_cudf/cudf/python/cudf/cudf/core/dataframe.py in select_dtypes(self, include, exclude)
   6612                 if is_categorical_dtype(i_dtype):
   6613                     include_subtypes.add(i_dtype)
-> 6614                 elif issubclass(dtype.type, i_dtype):
   6615                     include_subtypes.add(dtype.type)
   6616 

TypeError: issubclass() arg 1 must be a class
```
**Expected behavior**
The behavior we get without list type
```python
import cudf
df = cudf.DataFrame({'int_col':[0,1,2],'float_col':[1.1,2.2,3.3]})
df.select_dtypes('int64').columns
```
```python
Index(['int_col'], dtype='object')
``` 

**Environment overview (please complete the following information)**
 - Environment location: [Bare-metal]
 - Method of cuDF install: [conda]

**Environment details**
```python
cudf                      0.17.0a201203   cuda_10.2_py37_g5336301b85_372    rapidsai-nightly
libcudf                   0.17.0a201203   cuda10.2_g5336301b85_372    rapidsai-nightly
```

### PR DESCRIPTION
**Pull request by jpurviance**
Closes #6919 

Prevents select dtypes failure when list dyptes are in dataframe. Follows pandas API.

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

**Comment by shwina** [comment]
ok to test

**Comment by kkraus14** [comment]
ok to test

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: python/cudf/cudf/core/dataframe.py [hunk 1]
```diff
@@ -6741,7 +6741,7 @@ def select_dtypes(self, include=None, exclude=None):
                 # category handling
                 if is_categorical_dtype(i_dtype):
                     include_subtypes.add(i_dtype)
-                elif issubclass(dtype.type, i_dtype):
+                elif type(dtype.type) == type and issubclass(dtype.type, i_dtype):
```
Comment (shwina) [inline python/cudf/cudf/core/dataframe.py:6744]: How does this change exclude list types specifically?

**Comment by kkraus14** [comment]
Given this PR has gone stale I'm going to close it. @jpurviance if you're interested in updating this please open a new PR targeted at `branch-0.19`.

**Pull request closed without merging**